src/Microsoft.VisualStudio.Threading/ReentrantSemaphore.cs

```diff
@@ -274,19 +274,41 @@ internal NotRecognizedSemaphore(int initialCount, JoinableTaskContext joinableTa
             {
                 Requires.NotNull(operation, nameof(operation));
 
+                // Note: this code is duplicated and not extracted to minimize allocating extra async state machines.
+                // For performance reasons, we want to minimize the number of Joins performed, and also keep the size of the
+                // JoinableCollection to a minimum. This also means awaiting on the semaphore outside of a JTF.RunAsync. This requires
+                // us to not ConfigureAwait(true) on the semaphore. However, that breaks our contract of resuming on the sync context
+                // you started on. To partially fix this, we will at least resume you on the main thread or thread pool.
+                AsyncSemaphore.Releaser releaser = default;
```

Comment:
> = default [](start = 48, length = 10)

Can you remove this initializer? We unconditionally assign a value to the field further down. If we stopped doing that or added a condition to it, it might be by accident and we'd want the compiler to call it out. This initial value doesn't offer us anything, and it would suppress that compiler warning.
Same everywhere else it appears.